app/main.py

```diff
@@ -1,3 +1,16 @@
-def cache(func):
-    # Write your code here
-    pass
+from typing import Callable
+
+
+def cache(func: Callable) -> None:
+    store = {}
+
+    def inner(*args) -> str:
+        if args in store:
+            print("Getting from cache")
+            return store[args]
+        else:
+            result = func(*args)
+            print("Calculating new result")
+            store[args] = result
+        return result
```

Comment:
Redundant variable `result`. We can go without it

Reply:
hope, i've handled this task finally)